contracts/oracles/WamplOracle.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IOracle} from "../interfaces/IOracle.sol";
+import {IWAMPL} from "../interfaces/IWAMPL.sol";
+import {IChainlinkAggregator} from "../interfaces/IChainlinkAggregator.sol";
+
+/**
+ * @title WamplOracle Oracle
+ *
+ * @notice Provides a value onchain from a chainlink oracle aggregator
+ */
+contract WamplOracle is IOracle {
+    /// @dev The output price has a 8 decimal point precision.
+    uint256 public constant PRICE_DECIMALS = 8;
+    // The address of the Chainlink Aggregator contract
+    IChainlinkAggregator public immutable amplEthOracle;
+    uint256 public immutable amplEthOracleDecimals;
+    IChainlinkAggregator public immutable ethUsdOracle;
+    uint256 public immutable ethUsdOracleDecimals;
+    IWAMPL public immutable wampl;
+    uint256 public immutable unitAmpl;
+    uint256 public immutable unitWampl;
+    uint256 public immutable stalenessThresholdSecs;
+
+    constructor(
+        address _amplEthOracle,
+        address _ethUsdOracle,
+        address _wampl,
+        uint256 _stalenessThresholdSecs
+    ) {
+        IChainlinkAggregator __amplEthOracle = IChainlinkAggregator(_amplEthOracle);
```

Comment:
no reason to grab casted version separately, I would just take in as `IChainlinkAggregator` - or you can just cast where you need to

Reply:
I'd generally try to avoid double-underscoring as well as it's quite hard to read